pkg/cmd/step/create/step_create_meta_pipeline.go

```diff
@@ -291,11 +267,46 @@ func (o *StepCreatePipelineOptions) getClientsAndNamespace() (tektonclient.Inter
 	return tektonClient, jxClient, kubeClient, ns, nil
 }
 
-func (o *StepCreatePipelineOptions) buildPullRevs(pullRefs string) (*prow.PullRefs, error) {
-	var pr *prow.PullRefs
-	var err error
-	if pullRefs != "" {
-		pr, err = prow.ParsePullRefs(pullRefs)
+// determineBranchIdentifier finds a identifier for the branch which is build by the specified pull ref.
+// The name is either an actual git branch name (eg master) or a synthetic names  like PR-<number> for single pull requests
+// or 'batch' for Prow batch builds.
+// The method makes the decision purely based on the Prow pull ref. At this stage we don't have the full ProwJov spec
+// available.
+func (o *StepCreatePipelineOptions) determineBranchIdentifier(pullRef prow.PullRefs) string {
+	prCount := len(pullRef.ToMerge)
+	var branch string
+	switch prCount {
+	case 0:
+		{
+			// no pull requests to merge, taking base branch name as identifier
+			branch = pullRef.BaseBranch
+		}
+	case 1:
+		{
+			// single pull request, create synthetic PR identifier
+			for k := range pullRef.ToMerge {
+				branch = fmt.Sprintf("PR-%s", k)
+				break
+			}
+		}
+	default:
+		{
+			branch = "batch"
+		}
+	}
+	log.Logger().Debugf("branch identifier for pull ref '%s' : '%s'", pullRef.String(), branch)
+	return branch
+}
+
+func (o *StepCreatePipelineOptions) determinePipelineKind(pullRef prow.PullRefs) string {
+	var kind string
+
+	prCount := len(pullRef.ToMerge)
+	if prCount > 0 {
+		kind = jenkinsfile.PipelineKindPullRequest
+	} else {
+		kind = jenkinsfile.PipelineKindRelease
```

Comment:
How about `feature` pipelines? Not used very often, but it is a supported option.

Reply:
I meant to bring this up. I have it on my list of open questions. I am not quite sure what `feature` pipelines mean. That said, I don't see how our current code ever allows anything else than `release` or `pull request` atm - https://github.com/jenkins-x/jx/blob/fe66ed51204667e37ed067f65a7ebc67f7c75bbb/pkg/cmd/controller/controller_pipelinerunner.go#L205

Not saying this is correct, I also feel I would like to have at least one other option. I am not fully behind the idea of each merge to master creating a release. I'd like a "master" build occurring when the merge to master happens and then somehow trigger a release build based on some other condition. But right now, it seems we have these two options.

Reply:
If we want we can bring the explicit pipeline kind option back and pass it through, but from how we currently integrate with Prow it does not really change anything.

Reply:
Yeah, the more I think about it, the more I'm leaning towards ignoring `feature` for now. We can revisit it later if we need to.